driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionNotCreateMongocryptdClientTest.java

```diff
@@ -0,0 +1,224 @@
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Stream;
+
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.AbstractClientSideEncryptionTest.cryptSharedLibPathSysPropValue;
+import static com.mongodb.client.Fixture.getMongoClientSettings;
+import static com.mongodb.client.unified.UnifiedClientEncryptionHelper.localKmsProviderKey;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+import static java.util.Collections.singletonMap;
+import static java.util.stream.Collectors.toMap;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+/**
+ * See
+ * <a href="https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/tests#bypass-creating-mongocryptd-client-when-shared-library-is-loaded">
+ * 20. Bypass creating mongocryptd client when shared library is loaded</a>.
+ */
+public abstract class AbstractClientSideEncryptionNotCreateMongocryptdClientTest {
+    @Nullable
+    private static final String CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE = cryptSharedLibPathSysPropValue().orElse(null);
+    private static final int DEFAULT_MONGOCRYPTD_PORT = MongoCryptHelper.createMongocryptdClientSettings(null)
+            .getClusterSettings().getHosts().get(0).getPort();
+    private static final String LOCAL_KMS_PROVIDER_ID = "local";
+    private static final Duration TIMEOUT = Duration.ofMillis(1_000);
+    private static final MongoNamespace KEY_VAULT_NAMESPACE = new MongoNamespace("keyvault", "datakeys");
+
+    private MongoClient client;
+    private MongoCollection<Document> collection;
+    private ConnectionTracker mongocryptdConnectionTracker;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        assumeTrue(CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE != null);
+        mongocryptdConnectionTracker = ConnectionTracker.start();
+        client = createMongoClient(MongoClientSettings.builder(getMongoClientSettings())
+                .autoEncryptionSettings(AutoEncryptionSettings.builder()
+                        .kmsProviders(singletonMap(LOCAL_KMS_PROVIDER_ID, singletonMap("key", localKmsProviderKey())))
+                        .keyVaultNamespace(KEY_VAULT_NAMESPACE.getFullName())
+                        .extraOptions(Stream.of(
+                                new SimpleImmutableEntry<>("cryptSharedLibPath", CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE),
+                                new SimpleImmutableEntry<>("mongocryptdURI", format("mongodb://%s:%d/db?serverSelectionTimeoutMS=%d",
```

Comment:
The prose description does not specify `serverSelectionTimeoutMS`, however it speeds up the test in a situation when the test fails.